modules/salmon.cfmod.pl

```diff
@@ -0,0 +1,156 @@
+#   (module adapted from kallisto.cfmod.pl                               #
+# This file is part of Cluster Flow.                                     #
+#                                                                        #
+# Cluster Flow is free software: you can redistribute it and/or modify   #
+# it under the terms of the GNU General Public License as published by   #
+# the Free Software Foundation, either version 3 of the License, or      #
+# (at your option) any later version.                                    #
+#                                                                        #
+# Cluster Flow is distributed in the hope that it will be useful,        #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of         #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
+# GNU General Public License for more details.                           #
+#                                                                        #
+# You should have received a copy of the GNU General Public License      #
+# along with Cluster Flow.  If not, see <http://www.gnu.org/licenses/>.  #
+##########################################################################
+
+# Module requirements
+my %requirements = (
+        'cores'         => '8',
+        'memory'        => '15G',
+        'modules'       => ['salmon', 'samtools'],
+        'references'=> 'salmon',
+        'time'          => sub {
+                my $cf = $_[0];
+                my $num_files = $cf->{'num_starting_merged_aligned_files'};
+                $num_files = ($num_files > 0) ? $num_files : 1;
+                return CF::Helpers::minutes_to_timestamp ($num_files * 2 * 60);
+        }
+);
+
+# Help text
+my $helptext = "".("-"x17)."\n Kallisto\n".("-"x17)."\n
+Salmon is a tool for quantifying the expression of transcripts using 
+RNA-seq data. Salmon uses new algorithms (specifically, coupling the 
+concept of quasi-mapping with a two-phase inference procedure) to 
+provide accurate expression estimates very quickly (i.e. wicked-fast) 
+and while using little memory. Salmon performs its inference using an 
+expressive and realistic model of RNA-seq data that takes into 
+account experimental attributes and biases commonly observed in real 
+RNA-seq data.\n\n";
+
+# Setup
+my %cf = CF::Helpers::module_start(\%requirements, $helptext);
+
+# MODULE
+# Check that we have a genome defined
+if(!defined($cf{'refs'}{'salmon'})){
+   die "\n\n###CF Error: No salmon transcriptome index found in run file $cf{run_fn} for job $cf{job_id}. Exiting.. ###";
+} else {
+    warn "\nPseudoaligning against $cf{refs}{salmon}\n\n";
+}
+
+open (RUN,'>>',$cf{'run_fn'}) or die "###CF Error: Can't write to $cf{run_fn}: $!";
+
+# Print version information about the module.
+warn "---------- Salmon version information ----------\n";
+warn `salmon --version`;
+warn "\n------- End of Salmon version information ------\n";
+
```

Comment:
I have just recently gone through all modules adding to this a bit, eg. see the [kallisto module](https://github.com/ewels/clusterflow/blob/master/modules/kallisto.cfmod.pl#L71-L78):

```perl
# Print version information about the module.
my $version = `kallisto version`;
warn "---------- Kallisto version information ----------\n";
warn $version;
warn "\n------- End of Kallisto version information ------\n";
if($version =~ /kallisto, version ([\d\.]+)/){
  warn "###CFVERS kallisto\t$1\n\n";
}
```

If you could add this in here, so that we get a `###CFVERS Salmon  v1.2.3` string, that'd be great. The summary module then picks these up and puts them into the summary e-mail. Also MultiQC reads them.